store/store.go

```diff
@@ -649,18 +785,44 @@ func (c *MongoStoreClient) GetDeviceData(p *Params) (StorageIterator, error) {
 			query := generateMongoQuery(p)
 			query["type"] = theType
 			opts := options.FindOne().SetProjection(removeFieldsForReturn).SetSort(bson.M{"time": -1})
-			result, resultErr := dataCollection(c).
-				FindOne(c.context, query, opts).
-				DecodeBytes()
-			if resultErr != nil {
-				if resultErr == mongo.ErrNoDocuments {
-					continue
+			// collections to search. stop at first collection that has data.
+			collections := []*mongo.Collection{
+				dataCollection(c),
+			}
+			if theType == "upload" {
+				// If type is "upload", try both the dataSetsCollection, then
+				// dataCollection as it may exist in both, or just the
+				// original as the migration happens.
+				collections = []*mongo.Collection{
+					dataSetsCollection(c),
+					dataCollection(c),
 				}
-				err = resultErr
-				break
 			}
+			for _, collection := range collections {
+				if collection.Name() == dataSetsCollectionName {
```

Comment:
Do we need these if/else blocks? If the type is not `upload` and we are querying `deviceDataSets`, the query won't return anything - there's no need to remove the existing filter (it could even match incorrect records).

`query["type"] = theType` is already set few lines above if we are querying `deviceData`.

Reply:
No, don't need these blocks, I keep forgetting the uploads will be migrated as is from the original collection without any changes to them.